compass/landice/tests/mismipplus/tasks.py

```diff
@@ -0,0 +1,96 @@
+import xarray as xr
+
+# Following from:
+#   compass/ocean/tests/global_ocean/tasks.py
+
+
+def get_ntasks_from_cell_count(config, at_setup, mesh_filename):
+    """
+    Computes `ntasks` and `min_tasks` based on mesh size (i.e. nCells)
+
+    Parameters
+    ----------
+    config : compass.config.CompassConfigParser
+        Configuration options for the test case the step belongs to
+
+    at_setup: bool
+        Whether this method is being run during setup of the step, as
+        opposed to at runtime
+
+    mesh_filename : str
+        A file containing the MPAS mesh (specifically `nCells`) if
+        `at_setup == False`
+
+    Returns
+    -------
+    ntasks : int
+        The target number of MPI tasks
+
+    min_tasks : int
+        The minimum number of tasks
+    """
+
+    if at_setup:
+        # get the desired resolution from config file
+        resolution = config.getfloat("mesh", "resolution")
+        # get the reference number of cells as 1000m resolution
+        ncells_at_1km_res = config.getfloat("mismipplus", "ncells_at_1km_res")
+        # approximate the number of cells for the desired resolution
+        cell_count = int(ncells_at_1km_res * (1000 / resolution)**2)
+        # TODO: Account for extra cells when gutter is requested
+    else:
+        # get cell count from mesh
+        with xr.open_dataset(mesh_filename) as ds:
+            cell_count = ds.sizes['nCells']
+
+    # read MPI related parameters from configuration file
+    cores_per_node = config.getfloat('parallel', 'cores_per_node')
+    max_cells_per_core = config.getfloat('mismipplus', 'max_cells_per_core')
+    goal_cells_per_core = config.getfloat('mismipplus', 'goal_cells_per_core')
+
+    # from Xylar: machines (e.g. Perlmutter) seem to be happier with ntasks
+    #             that are multiples of 4
+    min_tasks = max(1, 4 * round(cell_count / (4 * max_cells_per_core)))
+    ntasks = max(1, 4 * round(cell_count / (4 * goal_cells_per_core)))
+    # if ntasks exceeds the number of cores per node, round to the nearest
+    # multiple of `cores_per_node`.
+    if ntasks > cores_per_node:
+        ntasks = int(cores_per_node) * round(ntasks / cores_per_node)
+
+    return ntasks, min_tasks
+
+
+def _approx_cell_count(config):
```

Comment:
Is this function called anywhere?  Maybe I'm missing it somehow, but I don't see a call to it.

Reply:
Oh, thanks for catching that Matt! Your totally right that the function is never called, but it definitely should be. 

The plan was to replace [Lines 34-40](https://github.com/andrewdnolan/compass/blob/bc691ee5a0ed5d1a235953e6db512ed6ea914b22/compass/landice/tests/mismipplus/tasks.py#L34-L40) of that same file with the function call. 

While I flushed out the function, I never actually deleted the old Lines 34-40 and replaced them with the function call. Related to your comment below, I guess that also why the function never returned anything although it definitely should. 

Both of the those things have been fixed and I'll push the related commit in just a second.